client/snap_op.go

```diff
@@ -34,6 +35,17 @@ func (client *Client) InstallSnap(name, channel string) (string, error) {
 	return client.doAsync("POST", path, nil, body)
 }
 
+// SideloadSnap sideloads the snap with the given path, returning the UUID
+// of the background operation upon success.
+func (client *Client) SideloadSnap(path string) (string, error) {
```

Comment:
a todo about have a param for letting set "X-Allow-Unsigned" would be good (I think we don't need in the integration tests because they run in developer mode, also maybe the checks are broken, not sure)